packages/block-library/src/block/edit.native.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { Text, View } from 'react-native';
+import { partial } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Spinner, Disabled } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { BlockEditorProvider, BlockList } from '@wordpress/block-editor';
+import { compose } from '@wordpress/compose';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+
+class ReusableBlockEdit extends Component {
+	constructor( { reusableBlock } ) {
+		super( ...arguments );
+
+		this.setBlocks = this.setBlocks.bind( this );
+
+		if ( reusableBlock ) {
+			// Start in edit mode when we're working with a newly created reusable block
+			this.state = {
+				// Since edition is not supported yet isEditing is always false
+				isEditing: false,
+				title: reusableBlock.title,
+				blocks: parse( reusableBlock.content ),
+			};
+		} else {
+			// Start in preview mode when we're working with an existing reusable block
+			this.state = {
+				isEditing: false,
+				title: null,
+				blocks: [],
+			};
+		}
+	}
+
+	componentDidMount() {
+		if ( ! this.props.reusableBlock ) {
+			this.props.fetchReusableBlock();
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.reusableBlock !== this.props.reusableBlock &&
+			this.state.title === null
+		) {
+			this.setState( {
+				title: this.props.reusableBlock.title,
+				blocks: parse( this.props.reusableBlock.content ),
+			} );
+		}
+	}
+
+	setBlocks( blocks ) {
+		this.setState( { blocks } );
+	}
+
+	render() {
+		const { isSelected, reusableBlock, isFetching, settings } = this.props;
+		const { isEditing, title, blocks } = this.state;
+
+		if ( ! reusableBlock && isFetching ) {
+			return <Spinner />;
+		}
+
+		if ( ! reusableBlock ) {
+			return (
+				<Text>
+					{ __( 'Block has been deleted or is unavailable.' ) }
+				</Text>
+			);
+		}
+
+		let element = (
+			<BlockEditorProvider
+				settings={ settings }
+				value={ blocks }
+				onChange={ this.setBlocks }
+				onInput={ this.setBlocks }
+			>
+				<BlockList withFooter={ false } />
+			</BlockEditorProvider>
+		);
+
+		if ( ! isEditing ) {
+			element = <Disabled>{ element }</Disabled>;
+		}
+
+		return (
+			<View>
+				{ isSelected && <Text style={ styles.title }>{ title }</Text> }
```

Comment:
I tried to imitate the web version behavior.
Web version | Mobile version
------------ | -------------
<img src=https://user-images.githubusercontent.com/14905380/92964147-ee9e7580-f473-11ea-8139-20e3ecfb7324.png width=300> | <img src=https://user-images.githubusercontent.com/14905380/92964240-0f66cb00-f474-11ea-823f-504d8f998183.png width=300>